go-controller/pkg/node/gateway_udn.go

```diff
@@ -717,3 +767,83 @@ func addRPFilterLooseModeForManagementPort(mgmtPortName string) error {
 	}
 	return nil
 }
+
+func (udng *UserDefinedNetworkGateway) SetUDNNetworkAdvertised(networkName string, isUDNNetworkAdvertised bool) {
```

Comment:
I don't think you need this method. Just do this on `Reconcile`. You will have the information updated iin NetInfo.